src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -213,25 +221,43 @@ class EthService(
     Future.successful(Right(SubmitHashRateResponse(true)))
   }
 
+  def getMining(req: GetMiningRequest): ServiceResponse[GetMiningResponse] = {
+    val isMining = lastActive.updateAndGet(new UnaryOperator[Option[Date]] {
+      override def apply(e: Option[Date]): Option[Date] = {
+        e.filter { time => Duration.between(time.toInstant, (new Date).toInstant).toMillis < minerTimeOut }
+      }
+    }).isDefined
+    Future.successful(Right(GetMiningResponse(isMining)))
+  }
+
+  private def reportActive() = {
+    val now = new Date()
+    lastActive.updateAndGet(new UnaryOperator[Option[Date]] {
+      override def apply(e: Option[Date]): Option[Date] = {
+        Some(now)
+      }
+    })
+  }
+
   def getHashRate(req: GetHashRateRequest): ServiceResponse[GetHashRateResponse] = {
     val hashRates: Map[ByteString, (BigInt, Date)] = hashRate.updateAndGet(new UnaryOperator[Map[ByteString, (BigInt, Date)]] {
       override def apply(t: Map[ByteString, (BigInt, Date)]): Map[ByteString, (BigInt, Date)] = {
         removeObsoleteHashrates(new Date, t)
       }
     })
-
+    reportActive()
```

Comment:
Is `getHashRate` used by `ethminer`? I wasn't able to find if it's used, but If it isn't we might not need to `reportActive` when this RPC method is called, right? I might be missing something.